ONSdigital/census-rh-ui
# CR-1744 Displayable Address

Add line to invalidate session identity if starting fulfilment journey - will invalidate auth session if it exists
Includes unit tests
Split out of CR-1733

Signed-off-by: Paul-Joel <paul.joel@ons.gov.uk>

# Motivation and Context
Fixes the 'Displayable address not in sample attributes' issue, by killing the identity from the session at the start of 'enter-address' for fulfilments

# What has changed
One line addition to pop session identity if not an auth journey
Unit tests to confirm this resolves the issue, so the user will get the session 'timeout' page rather than a code error

# How to test?
Browser window 1: /en/start/ - Enter a UAC, wait at the confirm page
Browser window 2: /en/request/access-code/enter-address/ - Just load page, do not enter a postcode
Without fix:
Browser window 1: Select 'Yes' to confirm address and press continue - should cause browser window 1 to throw a 500 error - with service failed to build eq payload error
With fix:
Browser window 1: Select 'Yes' to confirm address and press continue - should cause the session 'timeout' page to be returned

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1744
